webroot/js/vocabulary/add.ctrl.js

```diff
@@ -59,6 +59,7 @@
                     vm.vocabularyAdded.unshift(data);
                     vm.data.text = '';
                     vm.isAdding = false;
+                    setInterval(function() { document.getElementById('vocabulary-item').focus(); }, 1);
```

Comment:
`focus()` doesn't seem to work without the 1 millisecond delay. :/

Reply:
Ok, this is weird. It works with a 0 millisecond delay too, just needs `setInterval` to be there.

Reply:
[AngularJS uses its own digest cycle](https://docs.angularjs.org/guide/scope#integration-with-the-browser-event-loop) and any manipulation done outside of AngularJS isn't visible to this digest cycle. That's why you have to artificially trigger a change so that AngularJS gets notified.

Furthermore using this approach it is now impossible to focus any other element (e.g. the language selection dropdown or the input field for the search query) because the set interval fires immediately and puts the focus back to the vocabulary input field.